app/models/section.rb

```diff
@@ -181,10 +188,14 @@ def all_editions
 
 private
 
-  attr_reader :slug_generator, :editions, :latest_edition
+  attr_reader :slug_generator, :latest_edition, :previous_edition
 
   def published_edition
-    most_recent_non_draft = editions.reject(&:draft?).last
+    most_recent_non_draft = if !latest_edition.draft?
+                              latest_edition
+                            elsif previous_edition && !previous_edition.draft?
+                              previous_edition
+                            end
```

Comment:
Minor: You mention in the commit note that you had to make some changes due to a Rubocop violation, but you had to change this method anyway, didn't you? Unless I've misunderstood, I might be tempted to remove that sentence from the commit note.